tests/utils.py

```diff
@@ -7,5 +7,8 @@
 needs_py310 = pytest.mark.skipif(
     sys.version_info < (3, 10), reason="requires python3.10+"
 )
+needs_py312 = pytest.mark.skipif(
+    sys.version_info < (3, 12), reason="requires python3.12+"
+)
```

Comment:
Tbh I would prefer adding that as a custom mark:

In `conftest.py`:
```python
import sys

def pytest_runtest_setup(item):
    for mark in item.iter_markers():
        if mark.name == 'py312' and sys.version_info < (3, 12):
            pytest.skip("Cannot run on python versin < 3.12")
        ...
```

This would change the test code to:

```python
import pytest

@pytest.mark.py312
def test_only_executed_on_python_312_and_above():
    assert sys.version_info >= (3, 12)